crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -62,6 +62,16 @@ def services
     end
   end
 
+  api :POST, "/api/upgrade/nodes", "Initiate the upgrade of all nodes"
+  api_version "2.0"
+  # This is gonna initiate the upgrade of all nodes.
+  # The method runs asynchronously, so there's a need to poll for the status and possible errors
+  def nodes
+    # FIXME: implement this as asynchronous call!
```

Comment:
I do not think we should when the call is supposed to be asynchronous. We should check for error states by subsequent status calls.